lib/subjects/fallback_selection.rb

```diff
@@ -2,30 +2,58 @@ module Subjects
   class FallbackSelection
     attr_reader :workflow, :limit, :options
 
+    def self.num_training_subjects_limit
+      ENV.fetch('FALLBACK_NUM_TRAINING_SUBJECTS_LIMIT', 10)
+    end
+
     def initialize(workflow, limit, options = {})
       @workflow, @limit, @options = workflow, limit, options
     end
 
     def any_workflow_data
-      any_workflow_data_scope
-        .limit(limit)
-        .pluck("set_member_subjects.subject_id")
-        .shuffle
+      if workflow.grouped
+        fallback_grouped_selection.shuffle
+      else
+        fallback_selection.shuffle
+      end
     end
 
     private
 
-    def any_workflow_data_scope
-      scope = workflow.set_member_subjects
-      if workflow.grouped
-        if subject_set_id = options[:subject_set_id]
-          scope = scope.where(subject_set_id: subject_set_id)
-        else
-          msg = "subject_set_id parameter missing for grouped workflow"
-          raise Subjects::Selector::MissingParameter.new(msg)
-        end
+    def fallback_grouped_selection
+      unless (subject_set_id = options[:subject_set_id])
+        raise Subjects::Selector::MissingParameter, 'subject_set_id parameter missing for grouped workflow'
       end
-      scope
+
+      workflow
+        .set_member_subjects
+        .where(subject_set_id: subject_set_id)
+        .limit(limit)
+        .pluck('set_member_subjects.subject_id')
+    end
+
+    def fallback_selection
+      (non_training_subject_ids | training_subject_ids)
+    end
+
+    def non_training_subject_ids
+      SetMemberSubject
```

Comment:
It looks like the original fallback selection code was filtering based on non retired and unseen for user. See [workflow.set_member_subjects](https://github.com/zooniverse/panoptes/blob/master/lib/subjects/fallback_selection.rb#L19) , if you click into the def'n you'll see that it returns either `select_non_retired_unseen_for_user` or `select_non_retired`. Is that something we want to include in this query here?

Reply:
I'm not sure this is the case, the `workflow.set_member_subjects` is a defined association on the workflow and doesn't filter those records on their retirement status at all, https://github.com/zooniverse/panoptes/blob/1fabe3c3ae29955d07c8fdb82b273381660bbcaa/app/models/workflow.rb#L19

Reply:
Ah, you're totally right. My bad, when I clicked on `workflow.set_member_subjects` in github it brought me to [SetMemberSubjectSelector#set_member_subjects](https://github.com/zooniverse/panoptes/blob/1fabe3c3ae29955d07c8fdb82b273381660bbcaa/lib/subjects/set_member_subject_selector.rb#L11) and I mistakenly assumed that was the method that was being called. Thanks for clarifying that 👍